backend/src/main/service/CachingService.ts

```diff
@@ -0,0 +1,71 @@
+import { RedisClient } from 'redis';
+import { CachingObject } from '../data/CachingObject';
+import { Promotion } from '../entity/Promotion';
+
+export class CachingService {
+  private client: RedisClient;
+
+  constructor(client: RedisClient) {
+    this.client = client;
+  }
+
+  async cacheLatLonValues(
+    placeId: string,
+    latitude: number,
+    longitude: number
+  ): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      const value: CachingObject = { lat: latitude, lon: longitude };
+      this.client.setex(
+        placeId,
+        2592000,
+        JSON.stringify(value),
+        (err: Error | null, data: string) => {
+          if (err) {
+            return reject(err);
+          }
+          return resolve(true);
+        }
+      );
+    });
+  }
+
+  async getLatLonValue(placeId: string): Promise<CachingObject> {
+    return new Promise((resolve, reject) => {
+      this.client.get(placeId, (err: Error | null, data: string | null) => {
```

Comment:
Just curious, how come the parameters are `(err: Error | null, data: string | null)`